src/main/java/org/motechproject/mots/service/AutomatedReportService.java

```
@@ -0,0 +1,34 @@
+package org.motechproject.mots.service;
+
+import java.util.List;
+import org.motechproject.mots.domain.AutomatedReportSettings;
+import org.motechproject.mots.dto.AutomatedReportSettingsDto;
+import org.motechproject.mots.mapper.AutomatedReportSettingsMapper;
+import org.motechproject.mots.repository.AutomatedReportSettingsRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class AutomatedReportService {
+
+  @Autowired
+  private AutomatedReportSettingsRepository automatedReportSettingsRepository;
+
+  private static final AutomatedReportSettingsMapper MAPPER =
+      AutomatedReportSettingsMapper.INSTANCE;
+
+  public void updateSettings(AutomatedReportSettingsDto settingsDto) {
+    AutomatedReportSettings automatedReportSettings =
+        automatedReportSettingsRepository.findOneByJobName(settingsDto.getJobName());
+    if (automatedReportSettings == null) {
```

I think you should throw an exception if it's null